packages/members-api/lib/repositories/member.js

```diff
@@ -329,11 +323,38 @@ module.exports = class MemberRepository {
             'last_seen_at'
         ]);
 
+        // Determine if we need to fetch the initial member with relations
+        const needsProducts = this._stripeAPIService.configured && data.products;
+        const needsNewsletters = memberData.newsletters || typeof memberData.subscribed === 'boolean';
+
+        // Build list for withRelated
+        const requiredRelations = [];
+        if (needsProducts) {
+            requiredRelations.push('products');
+        }
+        if (needsNewsletters) {
+            requiredRelations.push('newsletters');
+        }
+
+        // Fetch the member with relations if required
+        let initialMember = null;
+        if (requiredRelations.length > 0) {
+            initialMember = await this._Member.findOne({
+                id: options.id
+            }, {...sharedOptions, withRelated: requiredRelations});
+
+            // Make sure we throw the right error if it doesn't exist
+            if (!initialMember) {
+                throw new NotFoundError();
```

Comment:
I _think_ this needs a message for that you'll wanna use the `@tryghost/tpl` library, and use `messages` object pattern like here: https://github.com/TryGhost/Ghost/blob/main/core/server/api/canary/members.js#L11-L30

Reply:
Ah yes, I followed the pattern at https://github.com/SimonBackx/Ghost/blob/022c8c8e6905337ee50724eaf6590ed7753d86d0/core/server/models/base/plugins/crud.js#L196